.tekton/pull-request.yaml

```diff
@@ -0,0 +1,73 @@
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  generateName: on-pr-
+  namespace: application-service
+  annotations:
+    pipelinesascode.tekton.dev/on-event: "[pull_request]"
+    pipelinesascode.tekton.dev/on-target-branch: "[main]"
+    pipelinesascode.tekton.dev/max-keep-runs: "2"
+spec:
+  params:
+    - name: repo_url
+      value: "{{repo_url}}"
+    - name: revision
+      value: "{{revision}}"
+  pipelineSpec:
+    params:
+      - name: repo_url
+        type: string
+      - name: revision
+        type: string
+    tasks:
+      - name: fetchit
+        params:
+          - name: url
+            value: $(params.repo_url)
+          - name: revision
+            value: $(params.revision)
+        taskRef:
+          kind: ClusterTask
+          name: git-clone
+        workspaces:
+          - name: output
+            workspace: source
+      - name: build
+        runAfter:
+          - fetchit
+        taskSpec:
+          metadata: {}
+          steps:
+            - image: >-
+                registry.access.redhat.com/ubi8/go-toolset
+              name: build
+              resources: {}
+              script: |
+                #!/usr/bin/env bash
+                set -eux
+
+                ls -ltr 
+
+                make build
```

Comment:
Should we either:

1) Build the operator docker image instead of the operator binary here, or
2) Or, have a separate step to build the operator docker image?

Just want to make sure that any changes to the operator dockerfile also get included in PR builds 🙂

Reply:
Couple things I realized re: testing.

1) Can you add a step for running `make test` after building the Operator?
2) Will this replace our current [CI GitHub Action](https://github.com/redhat-appstudio/application-service/blob/main/.github/workflows/pr.yml)? If so, how will we manage uploads to Codecov?

Reply:
fyi:

you can do both (have the CI GH action managing codecov) and other stuff with pac, if you want to go full black you can snatch this code  : 

https://github.com/openshift-pipelines/pipelines-as-code/blob/main/.tekton/pull-request.yaml#L147

you just need to make sure to add the secret in the namespace with the codecov token

Reply:
> Have a separate step to build the operator docker image?

Yes, I'd actually go for a step to build the image, tag it appropriately and push it, so that one can test it by deploying it if needed.

For now, I kept it to `make build` to be able to verify that the entire works end to end :)